arch/RISCV/RISCVDisassembler.c

```diff
@@ -0,0 +1,53 @@
+#include "RISCVDisassembler.h"
+#include "RISCVDetails.h"
+
+#include "riscv_decode.gen.inc"
+#include "riscv_insn_mapping.gen.inc"
+#include "riscv_ast2str.gen.inc"
+
+bool riscv_get_instruction(csh handle, 
+                    const uint8_t *code, size_t code_len, MCInst *instr, 
+                    uint16_t *size, uint64_t address, void *info) {
+    cs_insn *insn = instr->flat_insn;
+
+    if (!riscv_fill_size(insn, code[0])) {
+        return false;
+    }
+    // TODO: add compressed 2-bytes instructions
+    if (insn->size == 2) {
+
+    } else if (insn->size == 4) {
+        struct ast instruction;
+        decode(&instruction, code[3] << 24 | code[2] << 16 | code[1] << 8 | code[0]);
```

Comment:
For these byte extractions we have `readBytes32` and others in `utils.c`. Please use those.